### PR TITLE
Cleanup current culture handling in the unmanaged runtime

### PR DESCRIPTION
**Pull request by jkotas**
Large portion of the current culture handling in the unmanaged runtime inherited from desktop has been no-op. The nativeInitCultureAccessors QCall that it used to depend on desktop got (almost) never called in CoreCLR.

- Delete resetting of current culture on threadpool threads. It was needed in desktop because of a very tricky flow of current culture between appdomains. It is superseded by the flowing the current culture via AsyncLocal in CoreCLR.
- Comment out fetch of managed current culture for unmanaged resource lookup. It has number of problems that are not easy to fix. We are not localizing the unmanaged runtime currently anyway, so it is ok to just comment it out.
- Fix the rest to call CultureInfo directly without going through Thread.CurrentThread

### CODE REVIEW

**Comment by jkotas** [comment]
cc @filipnavara @sdmaclea

**Comment by jkotas** [comment]
> We are not localizing the unmanaged runtime currently

@sdmaclea Is it your understanding as well? (Just want to double check that the plans have not changed.)

**Comment by jkotas** [comment]
Related discussion https://github.com/dotnet/coreclr/pull/21672#discussion_r244211638

**Comment by filipnavara** [comment]
> Delete resetting of current culture on threadpool threads. It was needed in desktop because of a very tricky flow of current culture between appdomains. It is superseded by the flowing the current culture via AsyncLocal in CoreCLR.

If that's the case you can also remove `ResetThreadCulture` from `CultureInfo` and drop the call to it from CoreRT.

**Comment by jkotas** [comment]
@dotnet-bot OSX10.12 x64 Checked Innerloop Build and Test
@dotnet-bot Ubuntu arm Cross Checked Innerloop Build and Test
@dotnet-bot Ubuntu arm Cross Checked no_tiered_compilation_innerloop Build and Test
@dotnet-bot Windows_NT x64 Release CoreFX Tests

File: src/vm/threads.cpp [hunk 1]
```diff
@@ -8715,31 +8654,21 @@ OBJECTREF Thread::GetCulture(BOOL bUICulture)
...
 
-    // Get the actual thread culture.
-    OBJECTREF pCurThreadObject = GetExposedObject();
-    _ASSERTE(pCurThreadObject!=NULL);
+    OBJECTREF pCurrentCulture;
+    if (bUICulture) {
+        // Call the Getter for the CurrentUICulture.  This will cause it to populate the field.
+        MethodDescCallSite propGet(METHOD__CULTURE_INFO__GET_CURRENT_UI_CULTURE);
+        ARG_SLOT retVal = propGet.Call_RetArgSlot(NULL);
+        pCurrentCulture = ArgSlotToObj(retVal);
```
Comment (filipnavara) [inline src/vm/threads.cpp]: This gets the `CurrentUICulture` for the calling thread, not the thread represented by the `Thread` object. It didn't really work correctly before, so in terms of functionality it's just as broken as before, but a comment would be nice.

File: src/vm/threads.cpp [hunk 2]
```diff
@@ -8963,16 +8892,17 @@ void Thread::SetCulture(OBJECTREF *CultureObj, BOOL bUICulture)
...
+    };
 
-    // Set the culture property on the thread.
-    THREADBASEREF pThreadBase = (THREADBASEREF)(pCurThreadObject);
-    CallPropertySet(bUICulture
-                    ? METHOD__THREAD__SET_UI_CULTURE
-                    : METHOD__THREAD__SET_CULTURE,
-                    (OBJECTREF)pThreadBase, *CultureObj);
+    // Make the actual call.
+    propSet.Call_RetArgSlot(pNewArgs);
```
Comment (filipnavara) [inline src/vm/threads.cpp]: Same here... sets the `Current[UI]Culture` for the calling thread, not the thread represented by the `Thread` object. A comment would be helpful.

**Comment by jkotas** [comment]
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test please
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test (Jit - TieredCompilation=0) please

**Comment by jkotas** [comment]
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test please
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test (Jit - TieredCompilation=0) please

**Comment by jkotas** [comment]
@filipnavara Thanks for the review!

**Comment by sdmaclea** [comment]
> > We are not localizing the unmanaged runtime currently

> @sdmaclea Is it your understanding as well? (Just want to double check that the plans have not changed.)

I believe that is correct.  There are no current plans to localize the unmanaged runtime.

@jkotas Thanks for the clean up

**Pull request merged**